### PR TITLE
Support `enabled: False` config for plugins with validators

### PR DESCRIPTION
**Pull request by NeonDaniel**
Refactor plugin load logic to respect plugins disabled in config
Adds unit test coverage for PHAL plugin load
Relates to https://github.com/NeonGeckoCom/NeonCore/issues/686

### CODE REVIEW

**Comment by JarbasAl** [comment]
> I see that disabling a plugin that has a validator method isn't working in upstream PHAL, sending a PR there

this was a per-plugin thing, not universal, but this is a good change!

**Comment by builderjer** [comment]
Does this mean that the plugin MUST contain `"enabled": true`?
If the "enabled" key is not in the config, won't it return False?

**Comment by NeonDaniel** [comment]
> Does this mean that the plugin config MUST contain `"enabled": true`? If the "enabled" key is not in the config, won't it return False?

Only if there isn't a validator (matching existing behavior).

If the `enabled` key is `False`, then the plugin won't try to validate or load. If `enabled` is unset, then it follows the existing behavior where it will try to load if EITHER:
- there is a validator method that returns true
- `enabled` config is set to `True`

**Comment by JarbasAl** [comment]
it should load if enabled is not set and validator doesnt exist, ie, by default instaling a plugin is all that is needed to load it

only admin plugins have the extra checks

File: ovos_PHAL/service.py [hunk 1]
```diff
@@ -73,9 +73,18 @@ def load_plugins(self):
                 continue
 
             config = self.user_config.get(name) or {}
+            if config.get("enabled") is False:
+                # Configuration explicitly disabled plugin
+                LOG.debug(f"PHAL plugin {name} disabled in configuration")
+                continue
```
Comment (mikejgray) [inline ovos_PHAL/service.py:79]: I may be missing something here, but if the config value `enabled` is explicitly set to `False`, won't this leave the `enabled` variable unbound in this method? Enabled isn't defined before this point and it's defined in all of the other previous paths.

Comment (NeonDaniel) [reply ovos_PHAL/service.py:79]: Yes, but `continue` here means we just move on to the next item in the list, so we won't reach any references to `enabled`

**Review by mikejgray — APPROVED** [review]

**Comment by NeonDaniel** [comment]
> it should load if enabled is not set and validator doesnt exist, ie, by default instaling a plugin is all that is needed to load it
> 
> only admin plugins have the extra checks
> 
> `enabled: false` should be respected universally and thats what i thought this PR was doing

Without (and with) this PR, the default is `enabled=False` if there is no validator. This PR just adds support for the `enabled` config param when there is a validator.

I'm not opposed to changing the default behavior to load plugins without validators though; this would match the behavior for skills where the default is to load anything that is installed. @mikejgray does this change seem good to you since you already approved the changes as they are now?

**Review by mikejgray — APPROVED** [review]

**Pull request merged**